web-console/src/views/load-data-view/load-data-view.tsx

```diff
@@ -1076,11 +1076,21 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
             <LearnMore href={getIngestionDocLink(spec)} />
           </Callout>
           {ingestionComboType ? (
-            <AutoForm
-              fields={getIoConfigFormFields(ingestionComboType)}
-              model={ioConfig}
-              onChange={c => this.updateSpecPreview(deepSet(spec, 'spec.ioConfig', c))}
-            />
+            <>
+              <AutoForm
+                fields={getIoConfigFormFields(ingestionComboType)}
+                model={ioConfig}
+                onChange={c => this.updateSpecPreview(deepSet(spec, 'spec.ioConfig', c))}
+              />
+              {deepGet(spec, 'spec.ioConfig.inputSource.properties.secretAccessKey.password') && (
+                <FormGroup>
+                  <Callout intent={Intent.WARNING}>
+                    Inlining the access key into the ingestion spec is dangerous as it might appear
+                    in server log files and can be seen by anyone accessing this console.
```

Comment:
Can we provide a recommendation on what a better alternative is (ie use the environment variable password provider instead)

Reply:
How would you word that message?

Reply:
@sthetland any suggestions here?

Reply:
That might be a maintenance headache as we add and remove PasswordProvider

Reply:
@maytasm3 are you worried that the environment password provider might go away in the future? Or that we might provide a better way in the future and this text will be out of sync.

The reason I thought of this was because if I landed on this page, and saw the warning as is - I'd feel like I was at a dead end without an idea of how to proceed

Reply:
@suneet-s Yes, that is correct. There is a drop down box above the field ("secret key type") that gives you other options. If you select that the warning will go away. I think that is user-friendly enough without recommendation

Reply:
Would something like this be future proof? ... maybe?

``Warning: This key will be visible to anyone accessing this console and may appear in server logs. For production scenarios, use of a more secure secret key type is strongly recommended. ``

Reply:
Would a good compromise be to say that - "When running in production, you should pick a more secure `secret key type` from the drop down above" 

The little `i` thing next to the option can point to the relevant docs which we will keep up to date as we add/ remove password providers.

Reply:
@suneet-s do you consider your comment as a blocker? Or can it be done in a follow-up?

Reply:
@jihoonson I don't know what the apache rules are on what qualifies for a blocker. I think the UX isn't great if I see a big scary warning without a way to remediate. I like @sthetland's suggestion above. If there are committers who are ok with this, I'm not going to veto.